### ISSUE TITLE
New Password should not be the same as old one

### ISSUE THREAD

**Issue by vaibhavchellani** [MetaMask/metamask-extension#2470]
While changing password i found that we can keep the new password same as the old one . This is not a bug but certainly not how the flow should work , we should alert the user with a message.

**Comment by Zanibas** [comment]
Noted!

**Comment by vaibhavchellani** [comment]
@Zanibas  can i work on this ?

**Comment by danfinlay** [comment]
You're welcome to work on this, if you can patch it on `master`, it will get into future releases.

Currently the way you would need to implement this would be to add an unlock attempt before a new vault create in the restore vault method. If the unlock is successful, the new vault creation should result in a re-lock, and an error that the password was the same.

Currently, we simply expose the `keychainController` method `createNewVaultAndRestore`, but to add this logic, we would need to define some custom behavior [here, where that method is exposed to the UI](https://github.com/MetaMask/metamask-extension/blob/master/app/scripts/metamask-controller.js#L349).

**Comment by vishva007** [comment]
Hi, was going through this . Can I look into this ?

I found a function tryUnlockMetamask (password) in /ui/actions.js . Can that function be used ?
(or) 
Should we be adding new code to metamask-controller.js to check if the current password opens and close it back again .

**Comment by lazaridiscom** [comment]
@danfinlay , 2 things here:

- as a so-called "power-user", I am highly annoyed but any software enforcing me the choice of password. Usually I bypass this "not the same password" by changing it twice, so I'm back to the same.

- this should be delayed, as things change anyway due to the ongoing redesign

**Comment by danfinlay** [comment]
I agree with @lazaridiscom here, this is a pretty low priority, and we're currently focused on stabilizing/documenting the `metamask-controller` and key management, not introducing new features carelessly.

You're welcome to try to make this change, but we will probably be slow to merge it, and we are actively changing that area of code, so it is probably best to just postpone working on this until we have reworked our account management, at least.
